database/connection.js

```diff
@@ -0,0 +1,97 @@
+const knex = require('knex')({
+  client: 'pg',
+  connection: {
+    host     : '127.0.0.1',
+    user     : process.env.USER,
+    database : 'jaguar-metro',
+    charset  : 'utf8'
+  }
+});
+
+const bookshelf = require('bookshelf')(knex);
+
+const Train = bookshelf.Model.extend({
+  tableName: 'TRAIN',
+  passengers: function() {
+    return this.hasMany(Passenger, ['PASSENGER.id']).through(TrainAndPassenger);
+  },
+  stations: function() {
+    return this.hasMany(Station, ['STATION.id']).through(StationAndTrain);
+  },
+  lastStation: function() {
+    return this.belongsTo(Station, ['LastStation'])
+  }
+});
+
+const Station = bookshelf.Model.extend({
+  tableName: 'STATION',
+  trains: function() {
+    return this.hasMany(Train, ['TRAIN.id']).through(StationAndTrain);
+  },
+  passengers: function() {
+    return this.hasMany(Passenger, ['PASSENGER.id']).through(StationAndPassenger);
+  },
+  currentTrain: function() {
+    return this.hasMany(Train, ['LastStation'])
+  },
+  tickets: function() {
+    return this.hasMany(Ticket, ['STATION.id'])
+  }
+})
+
+const Passenger = bookshelf.Model.extend({
+  tableName: 'PASSENGER',
+  trains: function() {
+    return this.hasMany(Train, ['TRAIN.id']).through(TrainAndPassenger);
+  },
+  stations: function() {
+    return this.hasMany(Station, ['STATION.id']).through(StationAndPassenger);
+  },
+  tickets: function() {
+    return this.hasMany(Ticket, ['PASSENGER.id'])
+  }
+})
+
+const Ticket = bookshelf.Model.extend({
+  tableName: 'TICKET',
+  passenger: function(){
+    return this.belongsTo(Passenger, ['PASSENGER.id'])
+  },
+  station: function(){
+    return this.belongsTo(Station, ['STATION.id'])
+  }
+})
+
+const StationAndTrain = bookshelf.Model.extend({
+  tableName: 'STATION_AND_TRAIN',
```

Comment:
These feel like join table names (STATION_AND_PASSENGER, etc) might be better as station_trains
could go either way, that's the naming convention I've seen.